src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/CustomMarshallerAttributeFixer.cs

```diff
@@ -93,42 +131,267 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
             }
         }
 
-        private static (List<string> missingMembers, List<Diagnostic> fixedDiagnostics) GetRequiredShapeMissingMemberNames(IEnumerable<Diagnostic> diagnostics)
+        private static (HashSet<string> missingMembers, List<Diagnostic> fixedDiagnostics) GetRequiredShapeMissingMemberNames(IEnumerable<Diagnostic> diagnostics)
         {
-            List<string> missingMemberNames = new();
+            HashSet<string> missingMemberNames = new();
             List<Diagnostic> requiredShapeDiagnostics = new();
             foreach (var diagnostic in diagnostics)
             {
                 if (diagnostic.Id == AnalyzerDiagnostics.Ids.CustomMarshallerTypeMustHaveRequiredShape)
                 {
                     requiredShapeDiagnostics.Add(diagnostic);
-                    if (diagnostic.Properties.TryGetValue(CustomMarshallerAttributeAnalyzer.MissingMemberNames.Key, out string missingMembers))
+                    if (diagnostic.Properties.TryGetValue(MissingMemberNames.Key, out string missingMembers))
                     {
-                        missingMemberNames.AddRange(missingMembers.Split(CustomMarshallerAttributeAnalyzer.MissingMemberNames.Delimiter));
+                        missingMemberNames.UnionWith(missingMembers.Split(MissingMemberNames.Delimiter));
                     }
                 }
             }
 
-            return (missingMemberNames, requiredShapeDiagnostics);
+            return (new HashSet<string>(missingMemberNames), requiredShapeDiagnostics);
         }
 
-        private static async Task<Document> AddMissingMembers(Document doc, SyntaxNode node, List<string> missingMemberNames, CancellationToken ct)
+        private static async Task<Solution> AddMissingMembers(Document doc, SyntaxNode node, HashSet<string> missingMemberNames, CancellationToken ct)
         {
-            var editor = await DocumentEditor.CreateAsync(doc, ct).ConfigureAwait(false);
-            var gen = editor.Generator;
+            var model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            var entryPointTypeSymbol = (INamedTypeSymbol)model.GetEnclosingSymbol(node.SpanStart, ct);
+
+            // TODO: Convert to use the IOperation tree once IAttributeOperation is available
+            var managedTypeSymbolInAttribute = GetManagedTypeInAttributeSyntax(node.GetLocation(), entryPointTypeSymbol);
+
+            bool isLinearCollectionMarshaller = ManualTypeMarshallingHelper.IsLinearCollectionEntryPoint(entryPointTypeSymbol);
+
+            ManualTypeMarshallingHelper.TryResolveManagedType(entryPointTypeSymbol, ManualTypeMarshallingHelper.ReplaceGenericPlaceholderInType(managedTypeSymbolInAttribute, entryPointTypeSymbol, model.Compilation), isLinearCollectionMarshaller, (_, _) => { }, out ITypeSymbol managedType);
```

Comment:
Please pull out `(_, _) => { }` as something that is named. It isn't clear to me why this exists or what is supposed to do.